fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/InstrumentedTransaction.java

```diff
@@ -38,70 +39,74 @@
  * Wrapper around {@link Transaction} that instruments certain calls to expose their behavior with
  * {@link FDBStoreTimer} metrics.
  */
+@API(API.Status.INTERNAL)
 public class InstrumentedTransaction extends InstrumentedReadTransaction<Transaction> implements Transaction {
 
     @Nullable
     protected ReadTransaction snapshot; // lazily cached snapshot wrapper
 
-    public InstrumentedTransaction(@Nonnull StoreTimer timer, @Nonnull Transaction underlying) {
-        super(timer, underlying);
+    public InstrumentedTransaction(@Nonnull StoreTimer timer, @Nonnull Transaction underlying, boolean enableAssertions) {
+        super(timer, underlying, enableAssertions);
     }
 
     @Override
     public void addReadConflictRange(byte[] keyBegin, byte[] keyEnd) {
-        underlying.addReadConflictRange(keyBegin, keyEnd);
+        underlying.addReadConflictRange(checkKey(keyBegin), checkKey(keyEnd));
     }
 
     @Override
     public void addReadConflictKey(byte[] key) {
-        underlying.addReadConflictKey(key);
+        underlying.addReadConflictKey(checkKey(key));
     }
 
     @Override
     public void addWriteConflictRange(byte[] keyBegin, byte[] keyEnd) {
-        underlying.addWriteConflictRange(keyBegin, keyEnd);
+        underlying.addWriteConflictRange(checkKey(keyBegin), checkKey(keyEnd));
     }
 
     @Override
     public void addWriteConflictKey(byte[] key) {
-        underlying.addWriteConflictKey(key);
+        underlying.addWriteConflictKey(checkKey(key));
     }
 
     @Override
     public void set(byte[] key, byte[] value) {
-        underlying.set(key, value);
+        underlying.set(checkKey(key), checkValue(value));
```

Comment:
AFAICT, key and value size limits are only imposed on `set` operations (and `mutate`). I think clear, clear range, the conflict range operations, and the read operations actually allow larger keys. I've validated this by playing around with the Python API in its shell, but maybe it's worth getting better validation.

Reply:
But, just because you can, should you?  Is there a case where this is legitimate and the server can make sense of the trailing bytes?

Reply:
There's one only mediumly contrived example that I can think of right now: if you have a key of exactly 10k bytes, and you want to get the next key in the database, you might perform a scan from `key + \x00`, which would have length 10,001. In theory, if you had the right logic, you can notice that it's hit the limit and only scan from `key[:9999] + str(key[9999] + 1)` (or something like that)